jira-slack-server-integration/jira-slack-server-integration-plugin/src/main/java/com/atlassian/jira/plugins/slack/service/listener/SlackEventHandlerService.java

```diff
@@ -185,13 +185,15 @@ public boolean handleMessage(@Nonnull final SlackIncomingMessage message) {
                 log.debug("Skipping link unfurling for issue {}", key);
                 continue;
             }
-
+            final boolean isProjectAutoConvertEnabled = projectConfigurationManager.isProjectAutoConvertEnabled(issue.getProjectObject());
             // user will get just one invite even if there are multiple issue references in the message
             boolean isUserAllowedToSeeIssue = true;
             if (!isSenderAllowedToSeeIssue(message.getChannelId(), slackUser, issue)) {
                 if (!isInvitationToUserSent && !slackUser.isPresent()) {
-                    inviteUserToConnectToSlack(message, issue);
-                    isInvitationToUserSent = true;
+                    if (isProjectAutoConvertEnabled) {
```

Comment:
Nit pick: could have added this as a third condition to the `if` above.

Reply:
Yes, it's better way to write less lines of code but i wanted to highlight the solution of the issue